extensions/dicom-microscopy/src/utils/loadAnnotation.js

```diff
@@ -0,0 +1,70 @@
+import getDicomWebClient from './dicomWebClient';
+
+export default function loadAnnotation({
+  microscopyService,
+  displaySet,
+  extensionManager,
+  servicesManager,
+}) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const { metadata } = displaySet;
+
+      const dicomMicroscopyModule = await microscopyService.importDicomMicroscopyViewer();
+
+      const client = getDicomWebClient({
+        extensionManager,
+        servicesManager,
+      });
+
+      const viewportId = servicesManager.services.viewportGridService.getActiveViewportId();
+      const managedViewers = microscopyService.getManagedViewersForViewport(viewportId);
+      const managedViewer = managedViewers[0];
+
+      client
+        .retrieveSeriesMetadata({
+          studyInstanceUID: metadata.StudyInstanceUID,
+          seriesInstanceUID: metadata.SeriesInstanceUID,
+        })
+        .then(retrievedMetadata => {
+          const annotations = retrievedMetadata.map(metadata => {
+            return new dicomMicroscopyModule.metadata.MicroscopyBulkSimpleAnnotations({
+              metadata,
+            });
+          });
+
+          annotations.forEach(async ann => {
```

Comment:
This will return before the annotations are actually loaded.  I don't know if that is a problem or not, but if there are enough annotations it could result in displaying partial annotations.
Using a for loop allows waiting till everything is loaded, or using Promise.all on a annotations.map allows for ensuring the loading is done by the time this function returns.
Generally the assumption on load functions is that only one is permitted to be started, and that the resulting promise is the same one no matter how many loads are done, returning immediately when the load had been completed.

Reply:
Updated. Thx.